### ISSUE TITLE
feat: cancellable actions on dockerode api call

### ISSUE THREAD

**Issue by axel7083** [containers/podman-desktop#4363]
### Is your enhancement related to a problem? Please describe

On some long task, we might want to be able to cancel some actions (pulling, building etc.)

### Describe the solution you'd like

The dockerode library provide some good material to do that, for instance they take in most of their api an AbortSignal (See https://github.com/apocas/dockerode/commit/323ba36402911a71a4e2fc9671f5968b065c9acf)

A new property could be added to the Task object, making them optionally cancellable. The TaskUI could then have a cancellable: boolean property, and display a nice button on the Task to allow the user to cancel it if needed.

The implementation to manage the cancellable is already present through the `CancellationTokenSource` and could be reused.

basically we could (probably in different PR):
- [x] Add an optional AbortController property to most of the functions in the `container-registry` (https://github.com/containers/podman-desktop/pull/4364)
- [ ] Creating Task for the stuff taking a lot of time to execute (https://github.com/containers/podman-desktop/pull/4344)
- [ ] Making the Task optionally cancellable / Add some basic ui for controlling it (https://github.com/containers/podman-desktop/pull/4369)

### Additional context

PS: the typing definitions for dockerode is a bit old I guess, and even if almost all endpoint support the abort signal, the options do not expose all of them (see https://github.com/DefinitelyTyped/DefinitelyTyped/pull/67061)

**Comment by axel7083** [comment]
Completed with https://github.com/containers/podman-desktop/pull/4364
